tests/cypress/integration/reportRepair/address.spec.js

```diff
@@ -1,42 +1,65 @@
 import {intercept_address_search} from '../../support/helpers';
 
+function setup_addresses_search(setup_addresses_API) {
+  setup_addresses_API();
+  cy.get('button').click().then(()=>{
+    cy.get('input.govuk-input').type('SW1A 2AA');
+    cy.get('button').click();
+  });
+  cy.get('[data-cy=SectionLoaded]', { timeout: 10000 }).then(($loadedSection) => {});
+}
+
 describe('address', () => {
   beforeEach(() => {
-    intercept_address_search();
     cy.visit('http://localhost:3000/report-repair/');
     cy.contains('Something else').click();
     cy.get('button').click();
     cy.contains('No').click();
-    cy.get('button').click().then(()=>{
-      cy.get('input.govuk-input').type('SW1A 2AA');
-      cy.get('button').click();
-    });
-    cy.get('[data-cy=SectionLoaded]', { timeout: 10000 }).then(($loadedSection) => {});
   });
 
-  it('displays the question', () => {
-    cy.contains('Where is the repair located?');
-  });
+  describe('Content and interaction', () => {
+    beforeEach(()=>{
+      setup_addresses_search(intercept_address_search);
+    });
 
-  it('contains a can\t find my address link',  () => {
-    cy.contains('I can\'t find my address').click();
-    cy.url().should('include', '/report-repair/not-eligible');
-  });
+    it('displays the question', () => {
+      cy.contains('Where is the repair located?');
+    });
+
+    it('contains a can\t find my address link',  () => {
+      cy.contains('I can\'t find my address').click();
+      cy.url().should('include', '/report-repair/not-eligible');
+    });
 
+    context('When a user doesn\'t select anything', ()=>{
+      it('an error should be shown',  () => {
+        cy.get('button').click()
+        cy.contains('Required');
+      });
+    });
 
-  context('When a user doesn\'t select anything', ()=>{
-    it('an error should be shown',  () => {
-      cy.get('button').click()
-      cy.contains('Required');
+    context('When a user selects an option', ()=>{
+      it('next page is shown',  () => {
+        cy.get('select').select('1 Downing Street, London, SW1A 2AA')
+        cy.get('button').click()
+        cy.url().should('include', '/report-repair/repair-location');
+      });
     });
+
   });
 
-  context('When a user selects an option', ()=>{
-    it('next page is shown',  () => {
-      cy.get('select').select('1 Downing Street, London, SW1A 2AA')
-      cy.get('button').click()
-      cy.url().should('include', '/report-repair/repair-location');
+  describe('API addresses with nulls', () => {
+    context('When API addresses contain \'nulls\' they are not displayed', () => {
+      it('address line 1 is null', () => {
+        setup_addresses_search(()=>intercept_address_search(1, 'SW1A 2AA', true))
+        cy.get('select').contains(/^London, SW1A 2AA$/)
+      });
+      it('address line 2 is null', () => {
+        setup_addresses_search(()=>intercept_address_search(1, 'SW1A 2AA', false, true));
```

Comment:
Is there a way to only set parameters that you want to change from the defaults?